File: web/src/services/recommendedGame.ts
import apiClient from './api'

export interface RecommendedGame {
  id: number
  name: string
  reply: string
  reason: string | null
  tag: string | null
  game_category: string | null
  game_image: string | null
  sort_value: number
  game_id: string
  game_name: string
  created_at: string
  updated_at: string
}

export interface RecommendedGameListResponse {
  items: RecommendedGame[]
  total: number
}

export const listRecommendedGames = async (
  q?: string,
  page = 1,
  page_size = 20
): Promise<RecommendedGameListResponse> => {
  const params = new URLSearchParams()
  if (q) params.set('q', q)
  params.set('page', String(page))
  params.set('page_size', String(page_size))
  const { data } = await apiClient.get(`/recommended-games?${params.toString()}`)
  return data
}

export const getRecommendedGame = async (id: number): Promise<RecommendedGame> => {
  const { data } = await apiClient.get(`/recommended-games/${id}`)
  return data
}

export const createRecommendedGame = async (meta: {
  name: string
  reply: string
  reason?: string
  tag?: string
  game_category?: string
  game_image?: string
  game_id: string
  game_name: string
}): Promise<RecommendedGame> => {
  const { data } = await apiClient.post('/recommended-games', meta)
  return data
}

export const updateRecommendedGame = async (
  id: number,
  meta: {
    name?: string
    reply?: string
    reason?: string
    tag?: string
    game_category?: string
    game_image?: string
    game_id?: string
    game_name?: string
  }
): Promise<RecommendedGame> => {
  const { data } = await apiClient.put(`/recommended-games/${id}`, meta)
  return data
}

export const deleteRecommendedGame = async (id: number): Promise<void> => {
  await apiClient.delete(`/recommended-games/${id}`)
}

File: crates/hiveweb/src/models/recommended_game.rs
//! Data model for the legacy recommended-game feature.
//!
//! Retained to decode existing rows and serve compatibility APIs.

#![allow(deprecated)]

use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[deprecated(note = "Legacy recommended-game model; retained for compatibility only")]
#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct RecommendedGame {
    // 推荐表ID
    pub id: i64,
    // 推荐列表时显示的标签
    pub tag: Option<String>,
    // 推荐列表时显示的名称
    pub name: String,
    // assistant 回复文本
    pub reply: String,
    // 游戏卡片：推荐理由
    pub reason: Option<String>,
    // 游戏卡片：分类
    pub game_category: Option<serde_json::Value>,
    // 游戏卡片：游戏封面
    pub game_image: Option<String>,
    pub sort_value: i32,
    // 游戏卡片：游戏id
    pub game_id: String,
    // 游戏卡片：游戏名称
    pub game_name: String,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}
